src/tests/JIT/Directed/Arrays/nintindexoutofrange.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+using System;
+using System.Runtime.CompilerServices;
+
+class NintIndexOutOfRangeTest
+{
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static void Stelem_Ref(object[] arr, nint i, Object value)
+        => arr[i] = value;
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static void LdElemATestHelper(ref object nothingOfInterest)
+    {}
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static void LdElemA(object[] arr, nint i)
+    {
+        LdElemATestHelper(ref arr[i]);
+    }
+
+    public static int Main()
+    {
+        long longIndex = ((long)1) << 32;
+        nint index = (nint)longIndex;
+        bool failed = false;
+
+        // On a 32bit platform, just succeed.
+        if ((long)index != longIndex)
```

Comment:
```suggestion
        if (sizeof(long) != sizeof(nint))
```

Nit: The above seems clearer to me.